model-engine/model_engine_server/infra/gateways/resources/k8s_resource_types.py

```diff
@@ -507,6 +507,7 @@ def get_endpoint_resource_arguments_from_request(
     main_env = []
     if isinstance(flavor, RunnableImageLike) and flavor.env:
         main_env = [{"name": key, "value": value} for key, value in flavor.env.items()]
+    main_env.append({"name": "AWS_PROFILE", "value": build_endpoint_request.aws_role})
```

Comment:
should we put a test for this (to test various merging logic with user-provided aws profiles)